src/tests/system/tests/test_gpo.py

```diff
@@ -0,0 +1,822 @@
+        2. Authenticate 'user', 'deny_user' and 'deny_user1' with both su and ssh
+    :expectedresults:
+        1. User authentication is successful
+        2. User authentication is unsuccessful
+    :customerscenario: True
+    """
+    ad.user("user").add()
+    user1 = ad.user("user1").add()
+    user2 = ad.user("user2").add()
+    deny_user1 = ad.user("deny_user1").add()
+    deny_user2 = ad.user("deny_user2").add()
+    group = ad.group("group").add().add_members([user2])
+    deny_group = ad.group("deny_group").add().add_members([deny_user2])
+
+    ad.gpo("site policy").add().policy(
+        {
+            "SeInteractiveLogonRight": [user1, group, ad.group("Domain Admins")],
+            "SeDenyInteractiveLogonRight": [deny_user1, deny_group],
+        }
+    ).link()
+
+    client.sssd.domain["ad_gpo_access_control"] = "enforcing"
+    client.sssd.start()
+
+    assert client.auth.ssh.password(username="user1", password="Secret123")
+    assert client.auth.su.password(username="user1", password="Secret123")
+    assert client.auth.ssh.password(username="user2", password="Secret123")
+    assert client.auth.su.password(username="user2", password="Secret123")
+    assert not client.auth.ssh.password(username="user", password="Secret123")
+    assert not client.auth.su.password(username="user", password="Secret123")
+    assert not client.auth.ssh.password(username="deny_user1", password="Secret123")
+    assert not client.auth.su.password(username="deny_user1", password="Secret123")
+    assert not client.auth.ssh.password(username="deny_user2", password="Secret123")
+    assert not client.auth.su.password(username="deny_user2", password="Secret123")
+
+
+@pytest.mark.importance("critical")
+@pytest.mark.topology(KnownTopology.AD)
+def test_gpo__is_set_to_enforcing_with_no_policy(client: Client, ad: AD):
+    """
+    :title: Group policy object host base access control is set to permissive
+    :description:
+        When ad_gpo_access_control is set to enforcing with no policy applied. It should
+        permit all users to login.
+    :setup:
+        1. Create the following users; 'user', 'user1', 'deny_user1'
+        2. Configure sssd.conf with 'ad_gpo_access_control' = 'permissive'
+        3. Start SSSD
+    :steps:
+        1. Authenticate 'user', 'user1' and 'deny_user1' with both su and ssh
+        2. Check logs to ensure that SSSD is in enforcing
+    :expectedresults:
+        1. All user authentication is successful
+        2. SSSD is in enforcing
+    :customerscenario: True
+    """
+    ad.user("user").add()
+    ad.user("user1").add()
+    ad.user("deny_user1").add()
+
+    client.sssd.domain["ad_gpo_access_control"] = "enforcing"
+    client.sssd.start()
+
+    assert client.auth.ssh.password(username="user1", password="Secret123")
+    assert client.auth.su.password(username="user1", password="Secret123")
+    assert client.auth.ssh.password(username="user", password="Secret123")
```

Comment:
Why two allowed users?

Reply:
ssh and su, they use different keys. Suppose you want that in a different test =D